pytorch_lightning/utilities/io.py

```diff
@@ -8,4 +8,47 @@ def load(path_or_url: str, map_location=None):
     if parsed.scheme == '':
         # local file
         return torch.load(path_or_url, map_location=map_location)
+    elif parsed.scheme == 's3':
+        return load_s3_checkpoint(path_or_url, map_location=map_location)
     return torch.hub.load_state_dict_from_url(path_or_url, map_location=map_location)
+
+
+def load_s3_checkpoint(checkpoint_path, map_location, **pickle_load_args):
+    from torch.serialization import _legacy_load
```

Comment:
why is `_legacy_load` required here? I think we should aim for `torch.load` here

Reply:
We could use `boto3` to download an S3 file, then use `torch.load`. Thoughts?